src/herder/HerderImpl.cpp

```diff
@@ -787,7 +787,14 @@ HerderImpl::triggerNextLedger(uint32_t ledgerSeqToTrigger)
 void
 HerderImpl::setUpgrades(Upgrades::UpgradeParameters const& upgrades)
 {
-    mUpgrades.setParameters(upgrades);
+    mUpgrades.setParameters(upgrades, mApp.getConfig());
+    persistUpgrades();
+}
+
+std::string
+HerderImpl::getUpgrades()
```

Comment:
This does not match setUpgrades signature.

I would expect setUpgrades/getUpgrades to have matching arguments/return type.
Maybe rename that to getUpgradesJson or make it return mUpgrades.getParameters() and let the caller to worry about calling .toJson()?

Reply:
good point - I'll make that change